catkin_tools/jobs/utils.py

```diff
@@ -50,42 +50,67 @@ def get_env_loaders(package, context):
     return sources
 
 
+def merge_envs(job_env, overlay_envs):
+    '''
+    In the merged/linked case of single env, this function amounts to a straight
+    assignment, but a more complex merge is required with isolated result spaces,
+    since a package's build environment may require extending that of multiple
+    other result spaces.
+    '''
+    merge_path_values = {}
+
+    for overlay_env in overlay_envs:
+        for key, values_str in overlay_env.items():
+            if key.endswith('PATH'):
```

Comment:
Ah, that's right thanks for the reminder and https://github.com/catkin/catkin_tools/issues/382#issuecomment-285739921.

I guess this isn't as bad since it really only applies to isolated devel or install. That at least gives the user a way out, i.e. use a merged option.

I'm still worried about the visibility of this issue, but I don't see an effective way to raise this concern with the user. We could compare each matching env variable with a whitelist and let them know with a notice that this might not be how you expect this variable to be handled. But again, I don't see how to aggregate these and give them to the user in a place that they'll likely see, like at the end of the program.

So I guess what I'm saying is, leave it unless you can think of a way to try and notify the users when something unexpected might be happening.